frontend/src/components/details/top_card.svelte

```diff
@@ -108,16 +109,18 @@
     </div>
   </div>
 
-  {#if providers}
-    {#if !providers.length}
+  {#if freeProviders || flatrateProviders}
+    {#if !freeProviders.length && !flatrateProviders.length}
       <div class="pt-5">
         <div class="badge badge-lg shadow-2xl">
           No providers in {$currentCountry}
         </div>
       </div>
     {:else}
       <div class="sm:flex sm:justify-center grid grid-cols-4 pt-5">
-        {#each providers as provider}
+        {#each [...new Map(freeProviders
+              .concat(flatrateProviders)
+              .map(item => [item["provider_id"], item])).values()] as provider}
```

Comment:
Could live in util, since it's a useful trick - Might also be easier quickly understand what's happening with the abstraction

Reply:
<3